### PR TITLE
Allow to disable the Services module

### PR DESCRIPTION
**Pull request by poncovka**
Check whether the Services DBus module is enabled before it is used.
It might be disabled in the Anaconda configuration file.

Related: rhbz#1834535

### CODE REVIEW

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

**Comment by poncovka** [comment]
The commit message was updated.

File: anaconda.py [hunk 1]
```diff
@@ -580,19 +580,8 @@ def _earlyExceptionHandler(ty, value, traceback):
...
-    from pyanaconda.modules.common.constants.services import SERVICES
-    from pyanaconda.core.constants import TEXT_ONLY_TARGET
-
-    services_proxy = SERVICES.get_proxy()
-
-    if not services_proxy.DefaultTarget and (anaconda.tui_mode or flags.usevnc):
-        log.debug("no default systemd target set & in text/vnc mode - setting multi-user.target.")
-        services_proxy.SetDefaultTarget(TEXT_ONLY_TARGET)
+    # Set up the default target.
+    startup_utils.initialize_default_target(text_mode=anaconda.tui_mode)
```
Comment (jkonecny12) [inline anaconda.py:584]: This should have better name. What target? It's fine here but not really self-explanation in the startup_utils with all the other stuff around. Please rename this to something like `initialize_systemd_default_target`?

File: anaconda.py [hunk 2]
```diff
@@ -722,14 +711,7 @@ def _earlyExceptionHandler(ty, value, traceback):
...
     # This should be done sooner and somewhere else once it is possible.
-    from pyanaconda.core.constants import SETUP_ON_BOOT_DEFAULT, SETUP_ON_BOOT_ENABLED
-    from pyanaconda.modules.common.constants.services import SERVICES
-    services_proxy = SERVICES.get_proxy()
-
-    if services_proxy.SetupOnBoot == SETUP_ON_BOOT_DEFAULT:
-        if  not flags.automatedInstall:
-            # Enable by default for interactive installations.
-            services_proxy.SetSetupOnBoot(SETUP_ON_BOOT_ENABLED)
+    startup_utils.initialize_setup_on_boot()
```
Comment (jkonecny12) [inline anaconda.py:714]: I would rather see here `initialize_first_time_setup` or something similar.

**Review by jkonecny12 — APPROVED** [review]
Looks good to me. Just a few ideas below.

File: anaconda.py (hunk 1 above)
Comment (poncovka) [reply anaconda.py]: Fixed.

File: anaconda.py (hunk 2 above)
Comment (poncovka) [reply anaconda.py]: Fixed.

**Review by jkonecny12 — APPROVED** [review]
Looks great to me now! Thanks!

**Review by M4rtinK — APPROVED** [review]
Looks good to me as well, thanks! :)

**Comment by poncovka** [comment]
Rebased.

**Comment by poncovka** [comment]
Ported in https://github.com/rhinstaller/anaconda/pull/3114.

**Pull request merged**